src/sentry/static/sentry/app/views/workBatchDetailsWaitingToBeMerged/workbatchDetails.jsx

```diff
@@ -4,102 +4,153 @@ import withOrganization from 'app/utils/withOrganization';
 import {connect} from 'react-redux';
 import {resourceActionCreators} from 'app/redux/actions/shared';
 import LoadingIndicator from 'app/components/loadingIndicator';
-import {WORK_BATCH, WORK_BATCH_DEFINITION, EVENTS} from 'app/redux/reducers/index';
+import PropTypes from 'prop-types';
+import DetailsForm from './detailsForm';
+
+import {
+  WORK_BATCH,
+  WORK_BATCH_DETAILS,
+  WORK_BATCH_DEFINITION,
+  EVENTS,
+} from 'app/redux/reducers/index';
 
 class WorkbatchDetails extends React.Component {
   constructor(props) {
     super(props);
     this.fetchStaticContentsFromWip = this.fetchStaticContentsFromWip.bind(this);
+    this.fetchDetailedContentFromWip = this.fetchDetailedContentFromWip.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.sendButtonClickedEvent = this.sendButtonClickedEvent.bind(this);
+    this.initCurrentFieldValues = this.initCurrentFieldValues.bind(this);
+    this.setFetched = this.setFetched.bind(this);
     const getWipWorkbatch = this.props.getWipWorkbatch;
     const org = this.props.organization;
-    Promise.all([getWipWorkbatch(org)]).then(this.fetchStaticContentsFromWip);
+    getWipWorkbatch(org)
+      .then(this.fetchStaticContentsFromWip)
+      .then(this.fetchDetailedContentFromWip)
+      .then(this.initCurrentFieldValues)
+      .then(this.setFetched);
+  }
+
+  setFetched() {
+    this.setState({fetched: true});
   }
 
   fetchStaticContentsFromWip() {
     const byIds = this.props.workBatchEntry.byIds;
     const arr = Object.values(byIds);
     let wipWorkbatch = arr[0];
-    this.props.getConfiguredStaticContents(
+    return this.props.getConfiguredStaticContents(
       this.props.organization,
       wipWorkbatch.cls_full_name
     );
   }
 
-  render() {
-    //TODO: merge this with files in the workBatchDetails folder
-    const buttonsFromConfig = [];
+  fetchDetailedContentFromWip() {
+    const byIds = this.props.workBatchEntry.byIds;
+    const arr = Object.values(byIds);
+    let wipWorkbatch = arr[0];
+    return this.props.getWorkBatchDetails(this.props.organization, wipWorkbatch.id);
+  }
+
+  fetchDefinitionsFromProps() {
     let workBatchDefinitionEntry = this.props.workBatchDefinitionEntry;
     if (workBatchDefinitionEntry == null) {
-      return <LoadingIndicator />;
+      return null;
     }
     let {byIds, detailsId} = workBatchDefinitionEntry;
     if (detailsId == null) {
+      return null;
+    }
+    let workDefinition = byIds[detailsId];
+    let {fields, buttons, id} = workDefinition;
+    if (fields === null || buttons === null || id === null) {
```

Comment:
This is fine but just FYI, in JS you could just do this as: if(!fields || !buttons || !id) since you can use any falsey type for truth checking

Reply:
Unless of course you want to differentiate between null and the empty string, in which case ignore me ;)

Reply:
Thanks for the comment, I appreciate that!